src/EventStore.Core/Services/Transport/Enumerators/Enumerator.AllSubscriptionFiltered.cs

```diff
@@ -71,462 +59,380 @@ public class AllSubscriptionFiltered : IAsyncEnumerator<ReadResponse> {
-											return;
-										}
-
-										foreach (var @event in completed.Events) {
-											var position = @event.OriginalPosition!.Value;
-
-											if (position > caughtUp) {
-												NotifyCaughtUp(Position.FromInt64(position.CommitPosition,
-													position.PreparePosition));
-												return;
-											}
-
-											Log.Verbose(
-												"Live subscription {subscriptionId} to $all:{eventFilter} enqueuing historical message {position}.",
-												_subscriptionId, _eventFilter, position);
-											if (!_channel.Writer.TryWrite(new ReadResponse.EventReceived(@event))) {
-												ConsumerTooSlow(@event);
-												return;
-											}
-										}
-
-										ReadHistoricalEvents(Position.FromInt64(
-											completed.NextPos.CommitPosition,
-											completed.NextPos.PreparePosition));
-										return;
+			private Task<TFPos> SubscribeToLive() {
+				var nextLiveSequenceNumber = 0UL;
+				var confirmationPositionTcs = new TaskCompletionSource<TFPos>();
 
-										void NotifyCaughtUp(Position position) {
-											Log.Verbose(
-												"Live subscription {subscriptionId} to $all:{eventFilter} caught up at {position} because the end of stream was reached.",
-												_subscriptionId, _eventFilter, position);
+				_bus.Publish(new ClientMessage.FilteredSubscribeToStream(Guid.NewGuid(), _subscriptionId,
+					new CallbackEnvelope(OnSubscriptionMessage), _subscriptionId,
+					string.Empty, _resolveLinks, _user,
+					_eventFilter, (int)_checkpointInterval, checkpointIntervalCurrent: 0));
 
-											caughtUpSource.TrySetResult(position);
-										}
+				return confirmationPositionTcs.Task;
 
-									case FilteredReadAllResult.Expired:
-										ReadHistoricalEvents(Position.FromInt64(
-											completed.CurrentPos.CommitPosition,
-											completed.CurrentPos.PreparePosition));
-										return;
+				void OnSubscriptionMessage(Message message) {
+					try {
+						if (message is ClientMessage.NotHandled notHandled &&
+						    TryHandleNotHandled(notHandled, out var ex))
+							throw ex;
 
-									case FilteredReadAllResult.AccessDenied:
-										Fail(new ReadResponseException.AccessDenied());
-										return;
+						switch (message) {
+							case ClientMessage.SubscriptionConfirmation confirmed:
+								long caughtUp = confirmed.LastIndexedPosition;
 
-									case FilteredReadAllResult.InvalidPosition:
-										Fail(new ReadResponseException.InvalidPosition());
-										return;
+								Log.Debug(
+									"Subscription {subscriptionId} to $all:{eventFilter} confirmed. LastIndexedPosition is {position:N0}.",
+									_subscriptionId, _eventFilter, caughtUp);
 
-									default:
-										Fail(ReadResponseException.UnknownError.Create(completed.Result));
+								confirmationPositionTcs.TrySetResult(new TFPos(caughtUp, caughtUp));
+								return;
+							case ClientMessage.SubscriptionDropped dropped:
+								Log.Debug(
```

Comment:
I believe we should switch it to `Verbose` here.

Reply:
same as [this comment](https://github.com/EventStore/EventStore/pull/4135#discussion_r1473614863), i'm open to the change, but i suggest we do it in another PR as it also affects the other subscription types